src/main/kotlin/DatastoreElasticsearchImpl.kt

```diff
@@ -0,0 +1,1545 @@
+                .setSize(maxHits)
+                .execute()
+                .actionGet()
+        val hits: MutableList<Map<String, Any>> = Lists.newArrayList()
+        response.hits.forEach(Consumer { hit: SearchHit -> hits.add(hit.sourceAsMap) })
+        return SearchResult(response.hits.totalHits.value, hits)
+    }
+
+    /*** RE-INDEXING  */
+    private fun getIdFnForType(securableObjectType: SecurableObjectType): Function<Any, String> {
+        return when (securableObjectType) {
+            SecurableObjectType.AssociationType -> Function { at: Any -> (at as AssociationType).associationEntityType.id.toString() }
+            SecurableObjectType.Organization -> Function { o: Any -> (o as Organization).id.toString() }
+            else -> Function { aso: Any -> (aso as AbstractSecurableObject).id.toString() }
+        }
+    }
+
+    override fun triggerEntitySetIndex(
+            entitySets: Map<EntitySet?, Set<UUID?>>,
+            propertyTypes: Map<UUID?, PropertyType?>
+    ): Boolean {
+        val idFn = Function { map: Any -> (map as Map<String?, EntitySet>)[ConductorElasticsearchApi.ENTITY_SET]!!.id.toString() }
+        val entitySetMaps = entitySets.entries.stream().map<Map<String, Any?>> { entry: Map.Entry<EntitySet?, Set<UUID?>> ->
+            val entitySetMap: MutableMap<String, Any?> = Maps.newHashMap()
+            entitySetMap[ConductorElasticsearchApi.ENTITY_SET] = entry.key
+            entitySetMap[ConductorElasticsearchApi.PROPERTY_TYPES] = entry.value.stream().map { key: UUID? -> propertyTypes[key] }.collect(
+                    Collectors.toList()
+            )
+            entitySetMap
+        }.collect(Collectors.toList())
+        return triggerIndex(
+                ConductorElasticsearchApi.ENTITY_SET_DATA_MODEL,
+                ConductorElasticsearchApi.ENTITY_SET_TYPE,
+                entitySetMaps,
+                idFn
+        )
+    }
+
+    override fun triggerOrganizationIndex(organizations: List<Organization>): Boolean {
+        val idFn = Function { org: Any ->
+            (org as Map<String?, Any>)[SerializationConstants.ID_FIELD].toString()
+        }
+
+        val organizationObjects = organizations.map {
+            getOrganizationObject(it)
+        }.toList()
+
+        return triggerIndex(
+                ConductorElasticsearchApi.ORGANIZATIONS,
+                ConductorElasticsearchApi.ORGANIZATION_TYPE,
+                organizationObjects,
+                idFn
+        )
+    }
+
+    override fun triggerSecurableObjectIndex(
+            securableObjectType: SecurableObjectType,
+            securableObjects: Iterable<*>
+    ): Boolean {
+        val indexName = indexNamesByObjectType[securableObjectType]
+        val typeName = typeNamesByIndexName[indexName]
+        return triggerIndex(indexName, typeName, securableObjects, getIdFnForType(securableObjectType))
+    }
+
+    //TODO: Seems dangerous and like we should delete?
+    fun clearAllData(): Boolean {
```

Comment:
@katherinebernstein @makosblade Seems removable? I will merge if this build succeeds so just submit another PR when you see this tomorrow.

Reply:
I thought we had already removed this 🤔

Reply:
Oh, it was removed from DatastoreElasticsearchImpl.java